### DISCUSSION TITLE
That is an amazing product and a game changer

### DISCUSSION THREAD

**Discussion by paule96** [Azure/static-web-apps#700] [Show and tell]
This service really enables smaller and bigger websites to have a solid base to get started. It has everything you need, a place to put your frontend and make it fast and easy availed. With the integration it offers a nice way to host a fast and saleable API too.
And that all for free in for even medium websites this should be enough. It's crazy how cheap it now is to host a website in azure now.

This was for me and @kirkone the reason to build [CZ.Azure.FileExchange](https://github.com/codez-one/CZ.Azure.FileExchange) on this platform. It was a nice experience over all for this little project. 
I mean even the integration with PRs is super nice. For example can just now everybody review our preview of the new design of the project [here](https://github.com/codez-one/CZ.Azure.FileExchange/pull/15).
It is crazy how simple this was.

Still there is a little bit todo in the deployment experience for more complex projects. Even for this project I reverse engineered the container `mcr.microsoft.com/appsvc/staticappsclient:stable` to write my own [PowerShell script](https://github.com/codez-one/CZ.Azure.FileExchange/blob/main/build/deploy.ps1) to deploy to azure static websites. The main point here is to split the build and deployment in two different things to have more controller over it. But there are already issue to this in this repo here, so this will be conceivably better in the feature. Overall is it a wonderful experience to use this young azure service.

And all the hard complexity is hidden for the customer. I mean even custom routing etc. is so easy. If I just think about the know how you need to configure routes for example in the azure web app service, this is much simpler.

Hope i can use this soon at work. :)

Hope this feedback is good to hear for the services team, which create this service. Nice job!

**Comment by laika1234** [comment]
@paule96 The powershell script to deploy to SWA is amazing. May I know how you reverse engineered the container and find out all this mechanism?

**Reply by miwebst** [reply]
Just to be clear, reverse engineering the container is not a recommended path forward. These API's will change overtime and we will not answer support requests that utilize custom deployment scripts. We are investigating local deployment options to make this easier.

**Reply by paule96** [reply]
@laika1234 to answer your question, I wrote a blog post about it https://www.codez.one/debuggen-von-http-request-eines-docker-containers/

It's only available in German but I think a good translation tool will make the job.

But still, this is not supported at all. So as the first comment is mentioned APIs are changing, so you maybe must do it multiple times. And it will be flacky in the most painful situation. So only do it if you don't need this script to much.

**Reply by laika1234** [reply]
@paule96 Thank you so much! Good to learn how to capture network trace for Docker Container. It's a great blog.

**Comment by crashAlgorithm** [comment]
Agreed, I'm loving the product so far. Especially the out-of-the-box oidc integrations.
